raiden/network/rpc/client.py

```diff
@@ -234,80 +217,70 @@ def __repr__(self):
 
     def block_number(self):
         """ Return the most recent block. """
-        return quantity_decoder(self.call('eth_blockNumber'))
+        return quantity_decoder(self.rpccall_with_retry('eth_blockNumber'))
 
-    def nonce(self, address):
-        if len(address) == 40:
-            address = unhexlify(address)
+    def nonce_needs_update(self):
+        if self.nonce_available_value is None:
+            return True
 
-        with self.nonce_lock:
-            initialized = self.nonce_current_value is not None
-            query_time = now()
+        now = time.time()
 
-            if self.nonce_last_update > query_time:
-                # Python's 2.7 time is not monotonic and it's affected by clock
-                # resets, force an update.
-                self.nonce_update_interval = query_time - self.nonce_update_interval
-                needs_update = True
+        # Python's 2.7 time is not monotonic and it's affected by clock resets,
```

Comment:
there is now a monotonic clock in py 3.5 and above. Should we switch to that while you're at it?
https://docs.python.org/3/library/time.html#time.monotonic

Reply:
what are the semantics of this clock? it should be walltime and not cpu cycles

Reply:
Not exactly sure, just found this comment the last time I was working on the client. This is the PEP: https://www.python.org/dev/peps/pep-0418/#time-monotonic